src/Textual/encodings/textpreprocessing.jl

```diff
@@ -0,0 +1,124 @@
+"""
+    TextEncoding() <: Encoding
+
+Encodes `Paragraph`s by applying various textual transforms.
+
+
+Encodes
+- `Paragraph` -> `Paragraph`
+
+"""
+struct Sanitize <: Encoding
+    tfms
+end
+
+function Sanitize()
+    base_tfms = [
+        replace_all_caps,
+        replace_sentence_case,
+        convert_lowercase,
+        remove_punctuations,
+        basic_preprocessing,
+        remove_extraspaces,
+    ]
+    return Sanitize(base_tfms)
+end
+
+function encodedblock(p::Sanitize, block::Paragraph)
+    return block
+end
+
+function encode(p::Sanitize, context, block::Paragraph, obs)
+    for tfm in values(p.tfms)
+        obs = tfm(obs)
+    end
+    obs
+end
+
+struct Tokenize <: Encoding
+    tfms
+end
+
+function Tokenize()
+    base_tfms = [
```

Comment:
Do we ever need multiple tokenizers? I'm not super familiar with it.

If yes, it should be the same as above, i.e. make this a `DEFAULT_TOKENIZERS`